js/src/common/components/Page.js

```diff
@@ -12,15 +12,18 @@ export default class Page extends Component {
 
     this.onNewRoute();
```

Comment:
I'd move that to the end of the function. This way everything in `oninit` would run before `onNewRoute`.